dsa/validate.go

```diff
@@ -0,0 +1,51 @@
+package dsa
+
+import (
+	"github.com/prebid/prebid-server/v2/exchange/entities"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+
+	"github.com/buger/jsonparser"
+)
+
+const (
+	// Required - bid responses without DSA object will not be accepted
+	Required = 2
+	// RequiredOnlinePlatform - bid responses without DSA object will not be accepted, Publisher is an Online Platform
+	RequiredOnlinePlatform = 3
+)
+
+// Validate determines whether a given bid is valid from a DSA perspective.
+// A bid is considered valid unless the bid request indicates that a DSA object is required
+// in bid responses and it the object happens to be missing from the specified bid.
+func Validate(req *openrtb_ext.RequestWrapper, bid *entities.PbsOrtbBid) (valid bool) {
+	if !dsaRequired(req) {
+		return true
+	}
+	if bid == nil || bid.Bid == nil {
+		return false
+	}
+	_, dataType, _, err := jsonparser.Get(bid.Bid.Ext, "dsa")
+	if dataType == jsonparser.Object && err == nil {
+		return true
+	} else if err != nil && err != jsonparser.KeyPathNotFoundError {
+		return true
```

Comment:
I noticed this line is not covered, is it possible to cover this, or it hard because of JSON?

Reply:
Yes I tried to add coverage but I'm having trouble forcing it down this path. I looked at other instances where we call `jsonparser.Get` and we don't have coverage for similar logic in those instances either.